frontend/src/metabase-lib/v1/expressions/compiler.ts

```diff
@@ -33,38 +24,35 @@ export function compileExpression({
   startRule,
   query,
   stageIndex,
-  resolve: shouldResolve = true,
+  resolver = fieldResolver({
+    query,
+    stageIndex,
+    startRule,
+  }),
 }: {
   source: string;
   startRule: StartRule;
   query: Lib.Query;
   stageIndex: number;
-  resolve?: boolean;
+  resolver?: Resolver | null;
 }): CompileResult {
   try {
     const { tokens } = lexify(source);
     const { root } = parse(tokens, { throwOnError: true });
     const compiled = compile(root);
-    const expression = applyPasses(compiled, [
```

Comment:
All these passes are now either superflouous, or done directly in the compiler, where we have enough context.